src/paging.md

```diff
@@ -0,0 +1,108 @@
+# Paging
+
+At the end of the last chapter, we did a lot of work that wasn’t actually
+writing kernel code. So let’s review what we’re up to:
+
+1. GRUB loaded our kernel, and started running it.
+2. We’re currently running in ‘protected mode’, a 32-bit environment.
+3. We want to transition to ‘long mode’, the 64-bit environment.
+4. In order to do that, we have to do some work.
+
+We’re on step four. More specifically, here’s what we have to do:
+
+1. Set up ‘paging’.
+2. Set up a ‘GDT’.
+3. Jump to long mode.
+
+This section covers step one. The next two will cover the other two steps.
+Afterwards, we’ll be ready to stop writing assembly and start writing Rust!
+
+
+> **By the way...**
+>
+> There’s something we’re going to skip here, which we’d want to do in a more
+> serious kernel: check to make sure that our hardware can actually do this!
+> We’re going to just assume that our ‘hardware’ can run in 64-bit mode, because
+> we’re running our OS in QEMU, which supports all of these operations. But if
+> we were to run our OS on a real 32-bit computer, it would end up crashing.
+> We could check that it’s possible, and then print a nice error message
+> instead. But, we won’t cover that here. It’s not particularly interesting, and
+> we know that it will never fail. But it might be something you want to explore
+> on your own, for extra credit.
+
+## Paging
+
+So, step one: set up ‘paging’. What is paging? Paging is a way of managing
```

Comment:
I feel like this an entire paragraph is too early. We don't really know what paging is and we're already discussing segmentation as an alternative. It might be better to move the motivational strategies up higher before talking about paging directly.